<?xml version="1.0"?>

<launch>
  <arg name="file_in"   default="benchmark"/>
  <arg name="_max_matches"            default="200"/>
  <arg name="_test_robustness"        default="false"/>
  <arg name="_bundle_adjustment_tol"  default="0.05"/>
  <arg name="_rpt2"                   default="6" />
  <arg name="_rpt3"                   default="10" />
  <arg name="file_out"  default="result_robust_$(arg _test_robustness)_mm_$(arg _max_matches)_batol_$(arg _bundle_adjustment_tol)_rpt2_$(arg _rpt2)_rpt3_$(arg _rpt3)"/>

  <include file="$(find boris_drone)/launch/global_params.xml" />
  <include file="$(find boris_drone)/launch/gui.xml" />
  <include file="$(find boris_drone)/launch/slam.xml" >
    <arg name="max_matches"            value="$(arg _max_matches)"/>
    <arg name="test_robustness"        value="$(arg _test_robustness)"/>
    <arg name="bundle_adjustment_tol"  value="$(arg _bundle_adjustment_tol)"/>
    <arg name="rpt2"                   value="$(arg _rpt2)"/>
    <arg name="rpt3"                   value="$(arg _rpt3)"/>
    <arg name="only_init"              value="true" />
  </include>

  <node name="boris_drone_manual_pose_estimation" pkg="boris_drone" type="manual_pose_estimation" output="screen"></node>
  <node pkg="rosbag" type="play"   name="player"   output="screen" args="-q /home/$(env USER)/bagfiles/$(arg file_in).bag"/>
  <node pkg="rosbag" type="record" name="recorder" output="screen" args="-O /home/$(env USER)/bagfiles/$(arg file_out).bag /pose_visual /pose_estimation /benchmark"/>
</launch>
